docs/en/Request-Processing.rst

```diff
@@ -20,19 +20,32 @@ interface, which consists of two methods:
    handler. This method contains the handler’s request processing logic,
    and returns an optional ``Response``.
 
+Additionally, request handler interfaces for each available request type (e.g.
+IntentRequest, LaunchRequest, SkillEnabledEventRequest) are available and
+provide a type safe way of writing handlers for a particular type of request.
+When using the base ``RequestHandler`` interface you must do a type check, then
+cast to work against a particular request type. The typed handlers eliminate
+this requirement and handle the type checking for you. For example, a handler
+implementing the ``LaunchRequestHandler`` interface will only be invoked if the
+incoming request is a ``LaunchRequest``.
+
+A list of the currently available type specific request handlers is available `here
+<https://github.com/alexa/alexa-skills-kit-sdk-for-java/tree/2.0.x/ask-sdk-core/src/com/amazon/ask/dispatcher/request/handler/impl>`_.
+
 The following example shows a request handler that is configured to
-handle the ``HelloWorldIntent``.
+handle the ``HelloWorldIntent``. As you can see, the handler implements the
+``IntentRequestHandler`` interface since we are going to be handling an intent request.
 
 .. code:: java
 
-   public class HelloWorldHandler implements RequestHandler {
```

Comment:
Hey shall we keep these code snippets for existing users to reference, as well as showing usage of typed request handler without type checking and casting, like directly retrieving intent from IntentRequestHandler `intentRequest.getIntent().getName()`, so that they can notice the changes and benefits.

Reply:
I don't want to reuse the existing code verbatim since it's not the pattern I want to encourage going forward. However I've updated the PR to include better examples of how you'd do a type check and cast in the generic handler vs. how it works when using a typed handler. This should help to illustrate the differences.